invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/components/requests.js

```diff
@@ -246,7 +246,14 @@ export const RequestToggleComponent = ({
   updateQueryFilters,
   userSelectionFilters,
 }) => {
-  const [open, setOpen] = useState(true);
+  const [open, setOpen] = useState(undefined);
+
+  useEffect(() => {
```

Comment:
I don't understand why this is needed. Can you please explain?

You know why I don't like effect? It is because it is hiding what it is really happening. This function will run on each `render`: do we want that? ping @kpsherva @zzacharo FIY

Reply:
Ah I see, well I am using functional cmp mainly because I see that all the overriden cmps are declared this way, if class components can be used I will change it as I agree it's much more readable, maybe @zzacharo knows if possible, otherwise I will give it a try

Reply:
For the record: After discussion decided to go for a class component that uses the withState, instad of a functional cmp overriding the Toggle cmp in RSK